---
title: Processing the Chromium dataset
author: Estella Yixing Dong
date: "November 10, 2023"
output:
  BiocStyle::html_document:
    titlecaps: false
    toc_float: true
bibliography: /ref.bib
---


```{r style, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
```

# Downloading the count data

We obtain a single-cell RNA sequencing dataset of Xenopus tail cells from @janesick2022high
We download and cache the count matrix and metadata using the `r Biocpkg("BiocFileCache")` package.

```{r, warning=FALSE}
library(BiocFileCache)
bfc <- BiocFileCache("raw_data", ask = FALSE)
# countmat <- bfcrpath(bfc,
#     file.path("https://zenodo.org/records/10076046/files",
#         "Chromium_FFPE_Human_Breast_Cancer_Chromium_FFPE_Human_Breast_Cancer_count_sample_filtered_feature_bc_matrix.h5"))

countmat <- bfcrpath(bfc,
    file.path("https://cf.10xgenomics.com/samples/cell-exp/7.0.1",
        "Chromium_FFPE_Human_Breast_Cancer_Chromium_FFPE_Human_Breast_Cancer/Chromium_FFPE_Human_Breast_Cancer_Chromium_FFPE_Human_Breast_Cancer_count_sample_filtered_feature_bc_matrix.h5"))

anno <- bfcrpath(bfc,
    file.path("https://cdn.10xgenomics.com/raw/upload/v1695234604/Xenium%20Preview%20Data",
        "Cell_Barcode_Type_Matrices.xlsx"))

## Chromium
# https://cf.10xgenomics.com/samples/cell-exp/7.0.1/Chromium_FFPE_Human_Breast_Cancer_Chromium_FFPE_Human_Breast_Cancer/Chromium_FFPE_Human_Breast_Cancer_Chromium_FFPE_Human_Breast_Cancer_count_sample_filtered_feature_bc_matrix.h5

## Visium
# https://cf.10xgenomics.com/samples/spatial-exp/2.0.0/CytAssist_FFPE_Human_Breast_Cancer/CytAssist_FFPE_Human_Breast_Cancer_filtered_feature_bc_matrix.h5

## Xenium 1
# https://cf.10xgenomics.com/samples/xenium/1.0.1/Xenium_FFPE_Human_Breast_Cancer_Rep1/Xenium_FFPE_Human_Breast_Cancer_Rep1_outs.zip

## Xenium 2
# https://cf.10xgenomics.com/samples/xenium/1.0.1/Xenium_FFPE_Human_Breast_Cancer_Rep2/Xenium_FFPE_Human_Breast_Cancer_Rep2_outs.zip

```

# Processing the read counts

We load the counts into memory.

```{r}
sce <- DropletUtils::read10xCounts(countmat)
colnames(sce) <- sce$Barcode
rownames(sce) <- rowData(sce)$Symbol
```

# Processing the annotation

Processing the per-cell annotation, and store as a column `Annotation` in the `colData()` of the SCE object.

```{r}
library(readxl)
celltype <- read_excel(anno, sheet = "scFFPE-Seq")

cells_overlap <- intersect(colnames(sce), celltype$Barcode)
sce <- sce[, cells_overlap]
celltype <- celltype[celltype$Barcode %in% cells_overlap, ]

colData(sce) <- as(merge(data.frame(colData(sce)), celltype), "DFrame")
```

# Saving for upload 

We save the annotated Chromium to file for upload to `r Biocpkg("ExperimentHub")`.

```{r}
repath <- file.path("XeniumPreprintBreastCancerData", "Chromium")
dir.create(repath, showWarnings=FALSE, recursive=TRUE)
saveRDS(sce, file=file.path(repath, "chromium.rds"))
```

# Session information

```{r}
sessionInfo()
```

# References